graph/encoding/dot/decode.go

```diff
@@ -118,7 +130,40 @@ func (gen *generator) addStmt(dst Builder, stmt ast.Stmt) {
 	case *ast.EdgeStmt:
 		gen.addEdgeStmt(dst, stmt)
 	case *ast.AttrStmt:
-		// ignore.
+		var n UnmarshalerAttr
+		var ok bool
+		var dst string
+		switch stmt.Kind {
+		case ast.KindGraph:
+			n, ok = gen.graphAttr.(UnmarshalerAttr)
```

Comment:
Mostly of a personal preference, but I would check for `nil` rather than type asserting on the `UnmarshalerAttr` interface. Likewise for `nodeAttr` and `edgeAttr`.

For the current code, I had to [consult the playground](https://play.golang.org/p/ChPMnJqLnh) to persuade myself that the current code was correct, since I thought that any value which may be assigned to `graphAttr, nodeAttr, edgeAttr UnmarshalerAttr` must implement the `UnmarshalerAttr` interface to be a valid assignment, and thus all type assertions would yield true. As the playground confirmed, the exception for this is of course the `nil` value.